src/metrics/include/metrics/metrics.hpp

```diff
@@ -15,24 +15,75 @@
 #ifndef HEADER_058638D3_7069_42DC_809F_5DB533FCFBA3_INCLUDED
 #define HEADER_058638D3_7069_42DC_809F_5DB533FCFBA3_INCLUDED
 
+#include <boost/filesystem.hpp>
 #include <chrono>
 #include <optional>
 #include <type_traits>
 #include <unordered_map>
 
 #include <gennylib/conventions.hpp>
+#include <gennylib/Node.hpp>
 
 #include <metrics/operation.hpp>
 #include <metrics/v1/passkey.hpp>
 
+
 namespace genny::metrics {
 
 /**
- * @namespace genny::metrics::v1 this namespace is private and only intended to be used by genny's
- * own internals. No types from the genny::metrics::v1 namespace should ever be typed directly into
- * the implementation of an actor.
+ * Class wrapping the logic involving metrics formats.
+ */
+class MetricsFormat {
+public:
+    enum class Format {
+        kCsv,
+        kCedarCsv,
+        kFtdc,
+        kCsvFtdc,
+    };
+
+    MetricsFormat() : _format{Format::kCsv} {}
+    
+    MetricsFormat(const Node& node) : _format{str_to_enum(node.to<std::string>())} {}
+
+    MetricsFormat(const std::string& to_convert) : _format{str_to_enum(to_convert)} {}
+
+    bool useGrpc() const {
+        return _format == Format::kFtdc || _format == Format::kCsvFtdc;
+    }
+
+    bool useCsv() const {
+        return _format == Format::kCsv || _format == Format::kCedarCsv ||
+            _format == Format::kCsvFtdc;
+    }
+
+    Format get() const {
+        return _format;
+    }
+
+private:
+    Format _format;
+    Format str_to_enum(const std::string& to_convert) {
```

Comment:
`strToEnum` (won't call out others but plz ensure it's camelCase throughout)

Reply:
Oof thanks for catching!